### ISSUE TITLE
subcommand parsing

### ISSUE THREAD

**Issue by theansweris39** [scallop/scallop#26]
Hi,

I have a conf with 2 subcommands:

```
class Conf(args: Array[String])
extends ScallopConf(args) {
  val connect = new Subcommand("connect") {
    val ip = opt[String]("ip", required = true)
    val port = opt[Int]("port", required = true)
    val local = opt[String]("local", required = true)
    val remote = opt[String]("remote", required = true)
  }

  val listen = new Subcommand("listen") {
    val ip = opt[String]("ip")
    val port = opt[Int]("port", required = true)
  }
```

If the application gets called with sub cmd 'listen' everything is ok. But with 'connect', the 2 options 'local' and 'remote' are tried to assign to the 'listen' sub cmd which ends in an exception:

```
Exception in thread "main" org.rogach.scallop.exceptions.UnknownOption: Unknown option 'remote'
    at org.rogach.scallop.Scallop$$anonfun$3.apply(Scallop.scala:92)
    at org.rogach.scallop.Scallop$$anonfun$3.apply(Scallop.scala:92)
    at scala.Option.getOrElse(Option.scala:108)
    at org.rogach.scallop.Scallop.parse(Scallop.scala:91)
    at org.rogach.scallop.Scallop.parse(Scallop.scala:52)
    at org.rogach.scallop.Scallop.org$rogach$scallop$Scallop$$parsed(Scallop.scala:154)
    at org.rogach.scallop.Scallop$$anonfun$isSupplied$5.apply(Scallop.scala:435)
    at org.rogach.scallop.Scallop$$anonfun$isSupplied$5.apply(Scallop.scala:434)
    at scala.Option.map(Option.scala:133)
    at org.rogach.scallop.Scallop.isSupplied(Scallop.scala:434)
    at org.rogach.scallop.Scallop$$anonfun$isSupplied$2.apply(Scallop.scala:431)
    at org.rogach.scallop.Scallop$$anonfun$isSupplied$2.apply(Scallop.scala:431)
    at scala.Option.map(Option.scala:133)
    at org.rogach.scallop.Scallop.isSupplied(Scallop.scala:431)
    at org.rogach.scallop.ScallopConf$$anonfun$opt$1.apply$mcZ$sp(ScallopConf.scala:124)
    at org.rogach.scallop.ScallopOption.isSupplied(ScallopOption.scala:23)
    at de.avm.lipsync.daemon.Runner$.main(Runner.scala:39)
    at de.avm.lipsync.daemon.Runner.main(Runner.scala)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:601)
    at com.intellij.rt.execution.application.AppMain.main(AppMain.java:120)
```

Adding the 2 options to listen avoids the exception...

**Comment by Rogach** [comment]
Sorry, I failed to reproduce the exception. Can you give me the exact argument list that causes the error?

The test case:

``` scala
object Conf extends ScallopConf(
    List("connect", "-i", "127.0.0.1", "-p", "4444", "-l", "aoeu", "-r", "asdf")) {

  val connect = new Subcommand("connect") {
    val ip = opt[String]("ip", required = true)
    val port = opt[Int]("port", required = true)
    val local = opt[String]("local", required = true)
    val remote = opt[String]("remote", required = true)
  }

  val listen = new Subcommand("listen") {
    val ip = opt[String]("ip")
    val port = opt[Int]("port", required = true)
  }
}
Conf.connect.ip() ==== "127.0.0.1"
Conf.connect.port() ==== 4444
Conf.connect.local() ==== "aoeu"
Conf.connect.remote() ==== "asdf"
```

**Comment by theansweris39** [comment]
The exception is thrown if u call e.g. 'Conf.listen.ip.isSupplied'.

--tested on a different machine with your code

**Comment by Rogach** [comment]
Yes, now it fails. It's a sure bug, I'll fix it ASAP.

**Comment by Rogach** [comment]
Sorry for the delay - my internet provider had some problem for half a day, so I couldn't push changes.

I released a new version - `0.6.3`, it should be available from Maven Central in one or two hours. Thanks for reporting the bug!
